bundles/org.openhab.binding.mail/src/main/java/org/openhab/binding/mail/internal/SMTPHandler.java

```diff
@@ -63,48 +68,82 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public void initialize() {
-        config = getConfigAs(SMTPConfig.class);
+        updateStatus(ThingStatus.UNKNOWN);
 
-        if (config.port == 0) {
-            if (config.security == ServerSecurity.SSL) {
-                config.port = 465;
-            } else {
-                config.port = 25;
-            }
+        SMTPConfig config = getConfigAs(SMTPConfig.class);
+        if (config.sender instanceof String confSender) {
+            sender = confSender;
+        }
+
+        Email mail = new SimpleEmail();
+        mail.setHostName(config.hostname);
+
+        int port = config.port != 0 ? config.port : config.security == ServerSecurity.SSL ? 465 : 25;
+        switch (config.security) {
+            case SSL:
+                mail.setSSLOnConnect(true);
+                mail.setSslSmtpPort(Integer.toString(port));
+                break;
+            case STARTTLS:
+                mail.setStartTLSEnabled(true);
+                mail.setStartTLSRequired(true);
+                mail.setSmtpPort(port);
+                break;
+            case PLAIN:
+                mail.setSmtpPort(port);
+        }
+
+        if (!(config.username.isEmpty() || config.password.isEmpty())) {
+            mail.setAuthentication(config.username, config.password);
         }
 
+        Session localSession;
+        try {
+            localSession = mail.getMailSession();
+            localSession.getTransport().connect();
+        } catch (EmailException | MessagingException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            return;
+        }
+
+        this.session = localSession;
         updateStatus(ThingStatus.ONLINE);
     }
 
+    @Override
+    public void dispose() {
+        Session localSession = session;
+        if (localSession != null) {
+            try {
+                localSession.getTransport().close();
+            } catch (MessagingException ignore) {
+            }
+            session = null;
+        }
+    }
+
     /**
      * use this server to send a mail
      *
      * @param mail the Email that needs to be sent
      * @return true if successful, false if failed
      */
     public boolean sendMail(Email mail) {
+        Session localSession = session;
```

Comment:
If you do it like this a single failed login (or connection problem) during initialization leads to an "forever unable to send mail" state. I don't think that's really a good idea.